src/Feature/SideBar/Feature_SideBar.re

```diff
@@ -91,3 +98,25 @@ let toggle = (pane, state) =>
   } else {
     {...state, shouldSnapShut: true, isOpen: true, selected: pane};
   };
+
+let setDefaultLocation = (state, setting) => {
+  let location =
+    switch (setting) {
```

Comment:
You may have intentionally made the `"left"` case explicit, but it can be reduced to:

```reason
switch (setting) {
| "right" => Right
| _ => Left
}
```

Or indeed even just `let location = setting == "right" ? Right : Left;`

Reply:
I did purposefully do this explicitly. I'm a big fan of state machines, enums, etc, and while it's probably true that "not right" will always mean "left", they aren't truly opposites. There's nothing to say there couldn't be a future where there's "top" or "bottom" variants (though very, very unlikely).

All that to say, I did this on purpose, but can change if it's bothersome to project owners.

Reply:
I don't have a strong feeling on this particular case - thanks for checking!

In general, we try to use exhaustive checks - when possible - but that's certainly not the case for strings.